radix-transactions/src/validation/transaction_validator.rs

```diff
@@ -190,14 +200,23 @@ impl TransactionValidator {
     ) -> Result<ValidatedNotarizedTransactionV1, TransactionValidationError> {
         self.validate_intent_v1(&transaction.signed_intent.intent)?;
 
-        let encoded_instructions = Rc::new(manifest_encode(
-            &transaction.signed_intent.intent.instructions.inner.0,
-        )?);
+        self.check_reference_limits(vec![
```

Comment:
MINOR: For better encapsulation, perhaps we should move `self.check_reference_limits` inside `validate_intent_v1`?

And arguably `check_signature_limits` should be in `validate_signatures_v1`?

Reply:
After looking into this a bit more, I think it's better to move the limits check up instead of inside `validate_intent_v1`. 

This helps reduce unnecessary, heavy validation work when things are not within limits.

Reply:
I'm comfortable with this - although I'm not entirely sure I agree.

* If we wanted to optimize for avoiding unnecessary work then the intent checks would be at prepare time.
* If we care about code clarity, putting intent checks in an intent validation method feels like we're less likely to miss these if we create other composable validations (e.g. if we add a method to validate a subintent or something).

That said, I agree that over-arching checks of pan-transaction limits (of e.g. references or total signatures) need to be at a higher layer, or passed down as bounds to constrain a lower layer.

So I'm happy to keep with this approach.